software/contrib/euclid.py

```diff
@@ -141,23 +141,45 @@ def __init__(self, script):
         @param script  The EuclideanRhythms script that owns this menu
         """
         self.script = script
+        self.first_render = True
+
+        self.generator_index = None
+        self.k2_percent = None
+        self.read_knobs()
+
+    def read_knobs(self):
+        """Read the current state of the knobs and return whether or the state has changed, requiring a re-render
```

Comment:
Missing word, "not".
```suggestion
        """Read the current state of the knobs and return whether or not the state has changed, requiring a re-render
```